calicoctl.py

```diff
@@ -843,6 +839,135 @@ def bgppeer_show(version):
         print "No BGP Peers defined."
 
 
+def container_ip_add(container_name, ip, version, interface):
+    """
+    Add an IP address to an existing calico networked container.
+
+    :param container_name: The name of the container.
+    :param ip: The IP to add
+    :param version: The IP version
+    :param interface: The name of the interface in the container.
+    
+    :return: None
+    """
+    address = check_ip_version(ip, version, IPAddress)
+
+    # The netns manipulations must be done as root.
+    enforce_root()
+
+    pool = get_pool_or_exit(address)
+
+    info = get_container_info_or_exit(container_name)
+    container_id = info["Id"]
+
+    # Check the container is actually running.
+    if not info["State"]["Running"]:
+        print "%s is not currently running." % container_name
+        sys.exit(1)
+
+    # Check that the container is already networked
+    try:
+        endpoint_id = client.get_ep_id_from_cont(hostname, container_id)
+        endpoint = client.get_endpoint(hostname, container_id, endpoint_id)
+    except KeyError:
+        print "Container is unknown to calico. Tell calico about the " \
+              "container before adding addresses."
+        sys.exit(1)
+
+    # From here, this method starts having side effects. If something
+    # fails then at least try to leave the system in a clean state.
+    if not client.assign_address(pool, ip):
+        print "IP address is already assigned in pool %s " % pool
+        sys.exit(1)
+
+    try:
+        old_endpoint = copy.deepcopy(endpoint)
```

Comment:
Refactor this to be a method on Endpoint --- e.g. `endpoint.copy()`  that saves the user from having to figure out what counts as a safe copy.